src/main/java/io/debezium/connector/yugabytedb/YugabyteDBValueConverter.java

```diff
@@ -916,20 +920,29 @@ protected Object convertArray(Column column, Field fieldDefn, YugabyteDBType ele
             }
             else if (data instanceof PgArray) {
                 try {
-                    final Object[] values = (Object[]) ((PgArray) data).getArray();
+//                  final Object[] values = (Object[]) ((PgArray) data).getArray();
+                    final Object[] values = getArrayElements(data);
                     final List<Object> converted = new ArrayList<>(values.length);
                     for (Object value : values) {
                         converted.add(elementConverter.convert(resolveArrayValue(value, elementType)));
                     }
                     r.deliver(converted);
                 }
-                catch (SQLException e) {
+                catch (Exception e) {
                     throw new ConnectException("Failed to read value of array " + column.name());
                 }
             }
         });
     }
 
+    private String[] getArrayElements(Object data) {
+        // This simple hack may not be feasible for 2D arrays
+        // TODO: add test for 2D arrays
+        String dataString = data.toString();
+        dataString = dataString.substring(1, dataString.length() - 1);
+        return dataString.split(",");
+    }
+
```

Comment:
Instead of this, can't we just resuse the method of pgArray/YBArray by passing the default delimeter.
Ideally it should be able to handle recursion 2d/3d arrays.